### PR TITLE
Some ClayDropdown fixes

### PR DESCRIPTION
**Pull request by carloslancha**

### CODE REVIEW

**Comment by matuzalemsteles** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by jbalsas** [comment]
Hey @matuzalemsteles, did you see anything weird here? I'm going to merge so we don't clash in other things, but go ahead with the review and let @carloslancha know of anything you find or questions you have so we can discuss it!

File: packages/clay-dropdown/src/ClayDropdown.js [hunk 1]
```diff
@@ -41,6 +41,15 @@ ClayDropdown.STATE = {
...
 	expanded: Config.bool().value(false),
 
+	/**
+	 * Icon to use in the trigger.
+	 * @instance
+	 * @memberof ClayDropdown
+	 * @type {?string}
+	 * @default undefined
+	 */
+	icon: Config.string(),
```
Comment (matuzalemsteles) [inline packages/clay-dropdown/src/ClayDropdown.js:51]: Why no using `symbol` and not `icon`?

Improve the API by passing `spritemap` into the `icon`. We generate a better consistency with the other components.
```javascript
icon: Config.shapeOf({
    spritemap: Config.string().required(),
    symbol: Config.string().value('caret-bottom')
}).required(),
```

Soy using:
```soy
{if $icon.symbol != 'caret-bottom'}
	{call ClayIcon.render}
	    {param spritemap: $icon.spritemap /}
            {param symbol: $icon.symbol /}
	{/call}
{/if}

{if $label}
	{$label}
{/if}

{if $icon.symbol == 'caret-bottom'}
	{call ClayIcon.render}
		{param spritemap: $icon.spritemap /}
		{param symbol: $icon.symbol /}
	{/call}
{/if}
```

Would it be less confusing what do you think?

Comment (matuzalemsteles) [reply packages/clay-dropdown/src/ClayDropdown.js:51]: For some reason, the `symbol` is not set to the default value when it is not passed.

Comment (jbalsas) [reply packages/clay-dropdown/src/ClayDropdown.js:51]: Hey @matuzalemsteles, this attempts to achieve exactly that. Check https://github.com/brianchandotcom/liferay-portal/pull/53237 to see how the components are used in the end. 

Having complex state hierarchies makes them way too difficult to use outside our simple scenario. We need to aim for more flat states and also remove prefixes as much as possible so the APIs are simpler. For instance, a dropdown will always show a `label` and an `icon`, no need to call that `triggerLabel` and `triggerIcon.symbol`.

Comment (matuzalemsteles) [reply packages/clay-dropdown/src/ClayDropdown.js:51]: Yeah, It's very more complexity. 😅

**Pull request merged**